var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -274,37 +272,47 @@ def cmake_args(self):
     #
     # export SPACK_USER_CACHE_PATH=/tmp/spack
     # export SPACK_DISABLE_LOCAL_CONFIG=true
-    def test(self):
-        if not self.spec.satisfies("@develop") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
-
-        tests = [
-            ["NlpMdsEx1.exe", "400", "100", "0", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "1", "-selfcheck"],
-            ["NlpMdsEx1.exe", "400", "100", "0", "-empty_sp_row", "-selfcheck"],
+
+    def run_hiop(self, raja):
+
+        if raja:
+            exName = "NlpMdsEx1Raja.exe"
+        else:
+            exName = "NlpMdsEx1.exe"
+
+        exe = os.path.join(self.prefix.bin, exName)
+        if not os.path.exists(exe):
+            raise SkipTest(f"{exName} does not exist in version {self.version}")
+
+        options = [
+            ["400", "100", "0", "-selfcheck"],
+            ["400", "100", "1", "-selfcheck"],
+            ["400", "100", "0", "-empty_sp_row", "-selfcheck"],
         ]
 
-        if "+raja" in self.spec:
-            tests.extend(
+        if raja:
+            options.extend(
                 [
-                    ["NlpMdsEx1Raja.exe", "400", "100", "0", "-selfcheck"],
-                    ["NlpMdsEx1Raja.exe", "400", "100", "1", "-selfcheck"],
-                    ["NlpMdsEx1Raja.exe", "400", "100", "0", "-empty_sp_row", "-selfcheck"],
+                    ["400", "100", "0", "-selfcheck"],
+                    ["400", "100", "1", "-selfcheck"],
+                    ["400", "100", "0", "-empty_sp_row", "-selfcheck"],
                 ]
             )
```

Comment:
What is the output from testing this?  Since it duplicates the existing options starting at line 287, you'll be re-running the same checks twice for both executables.